File: apps/shared/src/branding.ts
/**
 * Single source of truth for AgentX Workmate product identity — TypeScript side.
 *
 * Mirrors `branding.py` at the repo root. The desktop app, the web dashboard,
 * and the Ink TUI all import from here rather than hard-coding the product
 * name, so a rename stays a one-file edit per language.
 *
 * `tests/test_branding_consistency.py` parses this file and asserts every
 * value matches `branding.py` and `scripts/lib/branding.sh`, so the three
 * mirrors cannot drift.
 */

/** Full product name. Banner titles, About dialogs, installer windows. */
export const PRODUCT_NAME = 'AgentX Workmate'

/** Short form for tight spaces (status bars, response labels, bot names). */
export const SHORT_NAME = 'AgentX'

/** Legal entity that publishes the product. */
export const VENDOR_NAME = 'AstralX Technology'

/** Where users are told to send support requests. */
export const SUPPORT_EMAIL = 'kien.le@astralx.com.vn'

/** Brand glyph used as a decorative prefix in the CLI and TUI. */
export const BRAND_GLYPH = '⬡'

/** The command users type. Console-script entry point name. */
export const CLI_COMMAND = 'agentx'

/** Long-running gateway process entry point. */
export const GATEWAY_COMMAND = 'agentx-gateway'

/** Agent Client Protocol adapter entry point. */
export const ACP_COMMAND = 'agentx-acp'

/** Config/state directory under `$HOME` on POSIX (`~/.agentx`). */
export const CONFIG_DIR_POSIX = '.agentx'

/** Config/state directory under `%LOCALAPPDATA%` on Windows. */
export const CONFIG_DIR_WINDOWS = 'agentx'

/** Per-project instruction file, alongside AGENTS.md / CLAUDE.md. */
export const PROJECT_CONFIG_FILE = '.agentx.md'

/** Prefix for every environment variable the product reads or writes. */
export const ENV_PREFIX = 'AGENTX_'

/** Reverse-DNS application id. macOS bundle id, Windows AppUserModelId. */
export const APP_ID = 'com.agentx.workmate'

/** Custom URL scheme the desktop app registers (`agentx://…`). */
export const PROTOCOL_SCHEME = 'agentx'

/** Python distribution name. */
export const DIST_NAME = 'agentx-workmate'

/** Desktop application name. electron-builder productName / executableName. */
export const DESKTOP_APP_NAME = 'AgentX Workmate'

/**
 * Web presence. Not registered yet — an empty string means "no link", and
 * consumers must omit the surrounding UI rather than render a dead link.
 */
export const WEBSITE_URL = ''
export const DOCS_URL = ''

/**
 * Return the full environment variable name for a bare `name`.
 * `envName('HOME')` -> `'AGENTX_HOME'`.
 */
export const envName = (name: string): string => `${ENV_PREFIX}${name}`

File: branding.py
"""Single source of truth for AgentX Workmate product identity.

Every user-visible name, command, path segment, and identifier that carries
the product brand is defined here exactly once.  Nothing else in the tree
should hard-code the product name or the CLI command — import from this
module instead, so a future rename is a one-file edit.

Import-safe: standard library only, no intra-project imports, no side
effects.  ``hermes_constants`` imports this, and ``hermes_constants`` is
itself imported from everywhere, so this module must stay dependency-free.

Mirrored, deliberately, in two other places for the non-Python surfaces:

  * ``apps/shared/src/branding.ts``  — desktop / web / TUI (TypeScript)
  * ``scripts/lib/branding.sh``      — installers (POSIX shell)

``tests/test_branding_consistency.py`` asserts all three stay in sync, so a
value changed here without changing the mirrors fails CI.
"""

from __future__ import annotations

# ── Product identity ──────────────────────────────────────────────────
#: Full product name.  Banner titles, About dialogs, installer windows.
PRODUCT_NAME = "AgentX Workmate"

#: Short form for tight spaces (status bars, response labels, bot names).
SHORT_NAME = "AgentX"

#: Legal entity that publishes the product.  Package author, copyright
#: lines, Linux package maintainer, macOS bundle vendor.
VENDOR_NAME = "AstralX Technology"

#: Where users are told to send support requests.
SUPPORT_EMAIL = "kien.le@astralx.com.vn"

#: Brand glyph used as a decorative prefix in the CLI and TUI.
BRAND_GLYPH = "⬡"  # ⬡ WHITE HEXAGON

# ── Commands ──────────────────────────────────────────────────────────
#: The command users type.  Console-script entry point name.
CLI_COMMAND = "agentx"

#: Long-running gateway process entry point.
GATEWAY_COMMAND = "agentx-gateway"

#: Agent Client Protocol adapter entry point.
ACP_COMMAND = "agentx-acp"

# ── Filesystem layout ─────────────────────────────────────────────────
#: Config/state directory under ``$HOME`` on POSIX (``~/.agentx``).
CONFIG_DIR_POSIX = ".agentx"

#: Config/state directory under ``%LOCALAPPDATA%`` on Windows.
CONFIG_DIR_WINDOWS = "agentx"

#: Per-project instruction file, alongside AGENTS.md / CLAUDE.md.
PROJECT_CONFIG_FILE = ".agentx.md"

# ── Environment ───────────────────────────────────────────────────────
#: Prefix for every environment variable the product reads or writes.
#: Kept as a constant because ``tools/code_execution_tool`` filters the
#: child environment by this prefix at runtime.
ENV_PREFIX = "AGENTX_"

# ── Packaging identifiers ─────────────────────────────────────────────
#: Reverse-DNS application id.  macOS bundle id, Windows AppUserModelId.
APP_ID = "com.agentx.workmate"

#: Custom URL scheme the desktop app registers (``agentx://…``).
PROTOCOL_SCHEME = "agentx"

#: Python distribution name (``pyproject.toml`` ``[project].name``).
DIST_NAME = "agentx-workmate"

#: Desktop application name.  electron-builder ``productName`` and
#: ``executableName``; also the ``Application Support`` folder name.
DESKTOP_APP_NAME = "AgentX Workmate"

# ── Web presence ──────────────────────────────────────────────────────
# Not registered yet.  Empty string means "no link" — every consumer must
# treat these as optional and omit the surrounding UI when unset, rather
# than rendering a dead link.
WEBSITE_URL = ""
DOCS_URL = ""


def env(name: str) -> str:
    """Return the full environment variable name for a bare ``name``.

    ``env("HOME")`` -> ``"AGENTX_HOME"``.  Use this instead of writing the
    prefix inline so the prefix stays greppable and changeable.
    """
    return f"{ENV_PREFIX}{name}"


def config_dir_name() -> str:
    """Return the platform-appropriate config directory *name* (not path)."""
    import sys

    return CONFIG_DIR_WINDOWS if sys.platform == "win32" else CONFIG_DIR_POSIX
